test_elasticsearch/test_server/conftest.py

```diff
@@ -30,24 +30,30 @@
 ELASTICSEARCH_REST_API_TESTS = []
 
 
+def _create(elasticsearch_url, transport=None):
+    # Configure the client with certificates
+    kw = {}
+    if elasticsearch_url.startswith("https://"):
+        kw["ca_certs"] = CA_CERTS
+
+    # Optionally configure an HTTP conn class depending on
+    # 'PYTHON_CONNECTION_CLASS' env var
+    if "PYTHON_CONNECTION_CLASS" in os.environ:
+        kw["node_class"] = os.environ["PYTHON_CONNECTION_CLASS"]
+
+    if transport:
+        kw["transport_class"] = transport
+
+    # We do this little dance with the URL to force
+    # Requests to respect 'headers: None' within rest API spec tests.
+    return elasticsearch.Elasticsearch(elasticsearch_url, **kw)
+
+
 @pytest.fixture(scope="session")
 def sync_client_factory(elasticsearch_url):
     client = None
     try:
-        # Configure the client with certificates and optionally
-        # an HTTP conn class depending on 'PYTHON_CONNECTION_CLASS' envvar
-        kw = {"ca_certs": CA_CERTS}
-        if "PYTHON_CONNECTION_CLASS" in os.environ:
-            kw["node_class"] = os.environ["PYTHON_CONNECTION_CLASS"]
-
-        # We do this little dance with the URL to force
-        # Requests to respect 'headers: None' within rest API spec tests.
-        client = elasticsearch.Elasticsearch(elasticsearch_url, **kw)
-
-        # Wipe the cluster before we start testing just in case it wasn't wiped
-        # cleanly from the previous run of pytest?
-        wipe_cluster(client)
-
+        client = _create(elasticsearch_url)
```

Comment:
there might be a more pytestonic way to parametrize the fixture to optionally take a custom transport class, but I think this might be easier to understand?